x-pack/plugins/ml/public/maps/util.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FeatureCollection, Feature } from 'geojson';
+import { ESSearchResponse } from '../../../../../src/core/types/elasticsearch';
+import { formatHumanReadableDateTimeSeconds } from '../../common/util/date_utils';
+import type { MlApiServices } from '../application/services/ml_api_service';
+import { MLAnomalyDoc } from '../../common/types/anomalies';
+import type { SearchFilters } from './anomaly_source';
+
+export type MlAnomalyLayers = 'typical' | 'actual' | 'connected';
+
+export async function getResultsForJobId(
+  mlResultsService: MlApiServices['results'],
+  jobId: string,
+  locationType: MlAnomalyLayers,
+  searchFilters: SearchFilters
+): Promise<FeatureCollection> {
+  const { timeFilters } = searchFilters;
+  // Query to look for the highest scoring anomaly.
+  const body: any = {
+    query: {
+      bool: {
+        must: [{ term: { job_id: jobId } }, { term: { result_type: 'record' } }],
+      },
+    },
+    size: 1000,
```

Comment:
Should we limit to 1000 records only? Did you consider passing `size` as an argument for this method?

Reply:
This is just to keep consistent with the explorer page and everywhere else we're pulling anomalies results from 🤔 
Might be good to talk about the possibility of passing that in as an argument. Probably for a follow-up though.